packages/components/src/switch/src/SwitchField.tsx

```diff
@@ -24,50 +24,69 @@ const SwitchToDescriptionSizeAdapter: SizeAdapter<SwitchFieldProps["size"], Text
     md: "sm"
 };
 
-export interface SwitchFieldProps extends StyledSystemProps, BaseComponentProps {
+interface SwitchFieldRenderProps {
     /**
      * Whether the switch field is disabled.
      */
     isDisabled?: boolean;
+}
+
+export interface SwitchFieldProps extends TempBaseComponentProps<SwitchFieldRenderProps>, StyledSystemProps {
     /**
      * A switch field can vary in size.
      * @default "md"
      */
     size?: ResponsiveProp<"sm" | "md">;
+    /**
+     * Whether the switch field is disabled.
+     */
+    isDisabled?: boolean;
 }
 
 function SwitchField(props: SwitchFieldProps, ref: ForwardedRef<HTMLDivElement>) {
     [props, ref] = useContextProps(props, ref, SwitchFieldContext);
+
     const { stylingProps, ...ownProps } = useStyledSystem(props);
+
     const {
-        className,
-        children,
         isDisabled,
         size: sizeProp = "md",
-        style,
         slot,
         ...otherProps
     } = ownProps;
 
     const size = useResponsiveValue(sizeProp) ?? "md";
 
-    const classNames = clsx(
-        className,
+    const renderProps = useRenderProps({
+        ...otherProps,
+        values: {
+            isDisabled: isDisabled || false
+        }
+    });
+
+    const descriptionId = useId();
+
+    const className = clsx(
         GlobalSwitchFieldCssSelector,
         cssModule(
             styles,
             "hop-SwitchField",
             size
         ),
-        stylingProps.className
+        stylingProps.className,
+        renderProps.className
```

Comment:
maybe we should destructure className and style from renderProps since below you are merging renderProps, but style and className are still included.